subsys/shell/backends/shell_uart.c

```diff
@@ -176,16 +177,19 @@ static void uart_tx_handle(const struct device *dev, struct shell_uart_int_drive
 		return;
 	}
 
-	len = ring_buf_get_claim(&sh_uart->tx_ringbuf, (uint8_t **)&data,
+	avail = ring_buf_get_claim(&sh_uart->tx_ringbuf, (uint8_t **)&data,
```

Comment:
There is a corner case that will not work here. `ring_buf_get_claim` does not return all data remaining in the buffer but all continuous data so if there is `n` bytes at the end of ring buffer space and `m` bytes and the beginning it will return `n` and `m` on the next call. With the new approach TX will be disabled after writing `n` which `m` bytes are still pending.

You won't see it if UART implementation is using 1 byte FIFO (most UARTs may use that).

Reply:
I made a change to address this, such that the function will now loop until either no data is available from the ring buffer or the UART does not consume any of it. If no data is available, the code will then disable TX interrupts.

That should avoid the case where more than one ISR call is required to empty the ring buffer. The only issue I can see is if somehow the shell continuously produces more data and the UART is able to continue consuming it all, we could loop indefinitely? I'm not sure if that is realistic?